src/bin/gpfdist/gpfdist.c

```diff
@@ -3535,6 +3565,9 @@ static int request_set_transform(request_t *r)
 			r->trans.name = start + strlen(param);
 	}
 
+	if (! r->trans.name && opt.X)
+		r->trans.name = opt.X;
```

Comment:
What if:
(1) there is a transformation with name SOME_TRANSFORMATION and type=input.
(2) SOME_TRANSFORMATION is set to be default transformation for gpfdist.
(3) there is a WRITABLE external table without a transformation.

Will not writing to this table result in error here? As `transform_lookup` will fail to find SOME_TRANSFORMATION and type=output...
```
    /*
     * at this point r->trans.name is the name of the transformation requested
     * in the url and r->is_get tells us what kind (input or output) to look for.
     * attempt to look it up.
     */
	tr = transform_lookup(opt.trlist, r->trans.name, r->is_get ? 0 : 1, opt.V);
    if (! tr)
    {
        if (r->is_get)
        {
            gprintln(r, "reject invalid request from %s [%s %s] - unsppported input #transform",
                     r->peer, r->in.req->argv[0], r->in.req->argv[1]);
            http_error(r, FDIST_BAD_REQUEST, "invalid request (unsupported input #transform)");
        }
        else
        {
            gprintln(r, "reject invalid request from %s [%s %s] - unsppported output #transform",
                     r->peer, r->in.req->argv[0], r->in.req->argv[1]);
            http_error(r, FDIST_BAD_REQUEST, "invalid request (unsupported output #transform)");
        }
        request_end(r, 1, 0);
        return -1;
    }
```

BTW, there is a typo in "unsppported".

Reply:
This case is an expected behavior. We act the same as if transformation is specified by user on table level. User will get the same error if transformation is not found.
Not related, but interesting case, is that file can contain input and output rule with the same name.

As about a typo, we shouldn't fix this in current patch.

Reply:
Indeed, current implementation meets the requirements in the initial technical proposal.

But, thinking more widely:
The original request from the customer was related to input transformations. Because they had problems when loading data into GPDB. As far as I understand, they didn't want to have a default tranformation for WRITABLE external tables.

Now, with this change, I think there is a possibility that
 - at first launch customer will set up only 'input' default transformation.
 - after that the customer will face that work with WRITABLE tables (that have no transformation specified) is broken.
 - with our support, or by himself, the customer will realize that he needs to setup a dummy 'output' default tranformation to make WRITABLE tables working. And that will look like an unnecessary W/A (as he had no issues with writable tables at all).

I'm worried, that during this route, the customer may become a bit upset.

So, if we are staying with current approach for same default transormation for both input and output, let's at least highlight this aspect in the helpfile.

And please discuss this issue with the architect - maybe the technical proposal needs adjustment.